lib/cog/chat/message_metadata.ex

```diff
@@ -0,0 +1,6 @@
+defmodule Cog.Chat.MessageMetadata do
```

Comment:
Some documentation on this module in particular would be nice, emphasizing that (for now, at least) it's a bit of a grab-bag of "stuff" that only really applies to Slack, and not more broadly.

On further reflection, I wonder if we're perhaps over-architecting things by making this a Conduit-validated module, particularly given its grab-bag nature. Maybe it should just be a keyword list until a broader and more general use case becomes apparent?